src/redux/books/books.js

```diff
@@ -1,26 +1,42 @@
-import { createReducer } from '@reduxjs/toolkit';
-
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
 
-export const addBook = (payload) => ({
+const id = () => Math.floor((1 + Math.random()) * 0x10000)
+  .toString(16)
+  .substring(1);
+
+export const addBook = (book) => ({
   type: ADD_BOOK,
-  payload,
+  book,
 });
 
-export const removeBook = (payload) => ({
+export const removeBook = (itemId) => ({
   type: REMOVE_BOOK,
-  payload,
+  itemId,
 });
 
-const booksReducer = createReducer([], {
-  [ADD_BOOK]: (state, action) => {
-    state.push(action.payload);
+const booksReducer = (state = [
+  {
+    itemId: id(), title: 'Oliver Twist', author: 'Charles Dikens', category: 'sc-fi',
   },
-
-  [REMOVE_BOOK]: (state, action) => {
-    state.filter((payload) => payload.id !== action.payload.id);
+  {
+    itemId: id(), title: 'The World as Will and Representation', author: 'Arthur Schopenhauer', category: 'philosophy',
   },
-});
+  {
+    itemId: id(), title: 'Thus Spoke Zarathustra', author: 'Friedrich Nietzsche', category: 'philosophy',
+  },
+  {
+    itemId: id(), title: 'Les Misérables', author: 'Victor Hugo', category: 'historical fiction',
+  },
+], action = {}) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.book];
+    case REMOVE_BOOK:
+      return [...state.filter((book) => book.itemId !== action.itemId)];
+    default:
+      return state;
+  }
+};
```

Comment:
- [ ] You're also expected to create a component for category and write the similar functions and reducer for it such that when the `check status` button is clicked, `Under Construction` text would be displayed.
You can check project requirements to confirm this. 

![project-rqt](https://user-images.githubusercontent.com/26597432/217922647-dae26d59-1637-40f9-8c1a-7ef54af0d9a0.png)

Reply:
done